client/protocols/unlock.py

```diff
@@ -150,6 +155,18 @@ def interact(self, unique_id, case_id, question_prompt, answer, choices=None, ra
 
             else:
                 correct = True
+            tg_id = -1
+            misU_count = {}
+            printed_msg = ""
+
+            if not correct:
+                misU_count, tg_id, msg = self.GuidanceUtil.guidance_msg(unique_id,input_lines,
+                    self.access_token,self.hash_key,self.args.guidance)
+                printed_msg += msg
```

Comment:
Also, it's odd that this variable is called printed_msg, but it's not being printed (at least not here). If this is being printed in the `guidance_msg` method, I'd prefer that you either

* move the printing into `unlock.py` instead of in `guidance` (if there's only one print statement)
* OR change the variable `msg` to be `printed_msg`.